pootle/checks.py

```diff
@@ -0,0 +1,158 @@
+# Initially set to some minimums based on:
+# 1. Ubuntu 12.04LTS's version 2.8.4 (10.04LTS was too old for RQ)
+# 2. RQ requires >= 2.6.0, and
+# 3. Wanting to insist on at least the latest stable that devs are using i.e.
+#    2.8.* versions of Redis
+REDIS_MINIMUM_REQUIRED_VERSION = (2, 8, 4)
+
+
+@checks.register()
+def test_library_versions(app_configs, **kwargs):
+    from django import VERSION as django_version
+    from lxml.etree import LXML_VERSION as lxml_version
+    from translate.__version__ import ver as ttk_version
+
+    errors = []
+
+    if django_version < DJANGO_MINIMUM_REQUIRED_VERSION:
+        errors.append(checks.Critical("Your version of Django is too old.",
+            hint="Try pip install --upgrade Django",
+            id="pootle.E004",
+        ))
+
+    if lxml_version < LXML_MINIMUM_REQUIRED_VERSION:
+        errors.append(checks.Warning("Your version of lxml is too old.",
+            hint="Try pip install --upgrade lxml",
+            id="pootle.E005",
+        ))
+
+    if ttk_version < TTK_MINIMUM_REQUIRED_VERSION:
+        errors.append(checks.Critical("Your version of Translate Toolkit is too old.",
+            hint="Try pip install --upgrade translate-toolkit",
+            id="pootle.E006",
+        ))
+
+    return errors
+
+
+@checks.register()
+def test_optional_dependencies(app_configs, **kwargs):
+    errors = []
+
+    try:
+        import Levenshtein
+    except ImportError:
+        errors.append(checks.Warning(_("Can't find python-levenshtein package. Updating against"
+            " templates is faster with python-levenshtein."),
+            hint="Try pip install python-levenshtein",
+            id="pootle.W010",
+        ))
+
+    return errors
+
+
+@checks.register()
+def test_redis(app_configs, **kwargs):
+    from django_rq.queues import get_queue
+    from django_rq.workers import Worker
+
+    errors = []
+
+    try:
+        queue = get_queue()
+        workers = Worker.all(queue.connection)
+    except Exception as e:
+        conn_settings = queue.connection.connection_pool.connection_kwargs
+        errors.append(checks.Critical(_("Could not connect to Redis (%s)") % (e),
+            hint=_("Make sure Redis is running on %(host)s:%(port)s") % (conn_settings),
+            id="pootle.E001",
+        ))
+    else:
+        if not workers or workers[0].stopped:
+            # We need to check we're not running manage.py rqworker right now..
+            import sys
+            if len(sys.argv) > 1 and sys.argv[1] != "rqworker":
+                errors.append(checks.Warning(_("No RQ Worker running."),
+                    hint=_("Run new workers with manage.py rqworker"),
+                    id="pootle.W001",
+                ))
+
+        redis_version = queue.connection.info()["redis_version"].split(".")
+        if tuple(int(x) for x in redis_version) < REDIS_MINIMUM_REQUIRED_VERSION:
+            errors.append(checks.Warning(_("Your version of Redis is too old."),
```

Comment:
As I said three times now, critical checks are for stuff where we don't even want to allow pootle to run with that setup. "Outdated redis" is not part of that.